src/components/mgt-people-picker/mgt-people-picker.ts

```diff
@@ -0,0 +1,428 @@
+      return;
+    }
+    this._userInput = event.target.value;
+    if (event.target.value) {
+      window.setTimeout(() => {
+        this.loadPersonSearch(this._userInput);
+      }, 300);
+    } else {
+      event.target.value = '';
+      this._userInput = '';
+      this.people = [];
+    }
+  }
+
+  private onUserKeyDown(event: any) {
+    if (event.keyCode == 40 || event.keyCode == 38) {
+      this.handleArrowSelection(event);
+      event.preventDefault();
+    }
+    if (event.code == 'Tab' || event.code == 'Enter') {
+      this.addPerson(this.people[this.arrowSelectionCount], event);
+      event.target.value = '';
+      event.preventDefault();
+    }
+  }
+
+  private handleArrowSelection(event: any) {
+    if (this.people.length) {
+      //update arrow count
+      if (event.keyCode == 38) {
+        //up arrow
+        if (this.arrowSelectionCount > 0) {
+          this.arrowSelectionCount--;
+        } else {
+          this.arrowSelectionCount = 0;
+        }
+      }
+      if (event.keyCode == 40) {
+        //down arrow
+        if (this.arrowSelectionCount + 1 !== this.people.length && this.arrowSelectionCount + 1 < this.showMax) {
+          this.arrowSelectionCount++;
+        } else {
+          this.arrowSelectionCount = 0;
+        }
+      }
+
+      const peopleList = this.renderRoot.querySelector('.people-list');
+      //reset background color
+      for (let i = 0; i < peopleList.children.length; i++) {
+        peopleList.children[i].setAttribute('style', 'background-color: transparent ');
+      }
+      //set selected background
+      peopleList.children[this.arrowSelectionCount].setAttribute('style', 'background-color: #f1f1f1;');
+    }
+  }
+
+  private addPerson(person: MgtPersonDetails, event: any) {
+    if (person) {
+      this._personName = '';
+      this._duplicatePersonId = '';
+      let chosenPerson: any = person;
+      let filteredPersonArr = this._selectedPeople.filter(function(person) {
+        return person.id == chosenPerson.id;
+      });
+      if (this._selectedPeople.length && filteredPersonArr.length) {
+        this._duplicatePersonId = chosenPerson.id;
+      } else {
+        this._selectedPeople.push(person);
+        this.fireCustomEvent('selectionChanged', this._selectedPeople);
+
+        this.people = [];
+        this._userInput = '';
+        this._personName = '';
+        this.arrowSelectionCount = 0;
+      }
+    }
+  }
+
+  private async loadPersonSearch(name: string) {
+    this.isLoading = true;
+    if (name.length) {
+      name = name.toLowerCase();
+      let provider = Providers.globalProvider;
+      let peoples: any;
+      if (provider && provider.state === ProviderState.SignedIn) {
+        let client = Providers.globalProvider.graph;
+        //determine if group property is requested && if different group than previously requested
+        if (this.group && this.group != this.groupPeople) {
```

Comment:
refactor code to avoid duplicate code

Reply:
Updated the call to attribute changed callback:
58270a8